web/src/components/storage/ProposalDeviceSection.jsx

```diff
@@ -0,0 +1,208 @@
+    e.preventDefault();
+    if (device !== undefined) onSubmit(device);
+  };
+
+  return (
+    <Form id={id} onSubmit={submitForm}>
+      <DeviceSelector
+        selected={device}
+        devices={devices}
+        onChange={changeSelected}
+      />
+    </Form>
+  );
+};
+
+/**
+ * Allows to select the installation device.
+ * @component
+ *
+ * @callback onChangeFn
+ * @param {string} device - Name of the selected device.
+ *
+ * @param {object} props
+ * @param {string} [props.current] - Device name, if any.
+ * @param {StorageDevice[]} [props.devices=[]] - Available devices for the selection.
+ * @param {boolean} [props.isLoading=false] - Whether to show the selector as loading.
+ * @param {onChangeFn} [props.onChange=noop] - On change callback.
+ */
+const InstallationDeviceField = ({
+  current,
+  devices = [],
+  isLoading = false,
+  onChange = noop
+}) => {
+  const [device, setDevice] = useState(devices.find(d => d.name === current));
+  const [isFormOpen, setIsFormOpen] = useState(false);
+
+  const openForm = () => setIsFormOpen(true);
+
+  const closeForm = () => setIsFormOpen(false);
+
+  const acceptForm = (selectedDevice) => {
+    closeForm();
+    setDevice(selectedDevice);
+    onChange(selectedDevice);
+  };
+
+  /**
+   * Renders a button that allows changing selected device
+   *
+   * NOTE: if a device is already selected, its name and size will be used for
+   * the button text. Otherwise, a "No device selected" text will be shown.
+   *
+   * @param {object} props
+   * @param {StorageDevice|undefined} [props.current] - Currently selected device, if any.
+   */
+  const DeviceContent = ({ device }) => {
+    return (
+      <Button variant="link" isInline onClick={openForm}>
+        {device ? deviceLabel(device) : _("No device selected yet")}
+      </Button>
+    );
+  };
+
+  if (isLoading) {
+    return <Skeleton screenreaderText={_("Loading selected device")} width="25%" />;
+  }
+
+  const description = _("Select the device for installing the system.");
+
+  return (
+    <>
+      <div className="split">
+        <span>{_("Installation device")}</span>
+        <DeviceContent device={device} />
+      </div>
+      <Popup
+        title={_("Installation device")}
+        description={description}
+        isOpen={isFormOpen}
+      >
+        <If
+          condition={devices.length === 0}
+          then={_("No devices found.")}
+          else={
+            <InstallationDeviceForm
+              id="bootDeviceForm"
+              current={device}
+              devices={devices}
+              onSubmit={acceptForm}
+            />
+          }
+        />
+        <Popup.Actions>
+          <Popup.Confirm
+            form="bootDeviceForm"
+            type="submit"
+            isDisabled={devices.length === 0}
+          >
+            {_("Accept")}
+          </Popup.Confirm>
+          <Popup.Cancel onClick={closeForm} />
+        </Popup.Actions>
+      </Popup>
+    </>
+  );
+};
```

Comment:
Note: most probably, these internal components does not make sense now that the device selector it's in its own section. I.e., most of the code could be _merged_ into the main component. But it is out of the scope of this PR, especially taking into account that there are more planned changes ahead that could affect this component too. So, let's do it in a _re-factorization round_.